profiles/models.py

```diff
@@ -118,8 +118,15 @@ class SavedCompany(models.Model):
 
 class ViewedCompany(models.Model):
     date = models.DateTimeField(auto_now_add=True)
-    user = models.ForeignKey(CustomUser, on_delete=models.CASCADE)
+    user = models.ForeignKey(
+        CustomUser, on_delete=models.CASCADE, blank=True, null=True
+    )
     company = models.ForeignKey(Profile, on_delete=models.CASCADE)
 
-    class Meta:
-        unique_together = (("user", "company"),)
+    @property
+    def user_profile_name(self):
+        return self.user.profile.name
```

Comment:
This feature (wrapping some stuff from related models into properties) should be used very carefully.

When other developer will write code related to this model, autocompletion will show him that there is such field on the model. The problem is, that it's not a field on a model, its code which retrieves a chain of 2 other relations. 

So, it's 2 requests for DB, which are a bit obfuscated and it's easy to miss the need to make `select_related`/`prefetch_related`.